app/src/main/java/com/portfolio/tasky/entry/repositories/EntryRepositoryImpl.kt

```diff
@@ -0,0 +1,59 @@
+package com.portfolio.tasky.entry.repositories
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.portfolio.tasky.entry.network.EntryApiCall
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import com.portfolio.tasky.networking.usecases.domain.TaskyCallStatus
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import retrofit2.Response
+import javax.inject.Inject
+
+
+class EntryRepositoryImpl@Inject constructor(private val entryApiCall: EntryApiCall, private val taskyCallStatus: TaskyCallStatus) : EntryRepository  {
+    private val registrationLiveData = MutableLiveData<Boolean>()
+    private val loginLiveData = MutableLiveData<AuthenticationResponse?>()
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler{ _, throwable ->
+        throwable.printStackTrace()
+    }
+
+    override fun doRegistration(registerRequest: RegisterRequest): MutableLiveData<Boolean> {
+
+        CoroutineScope(Dispatchers.IO + coroutineExceptionHandler).launch {
+            taskyCallStatus.onRequestCallStarted()
+
+            val response : Response<Void> = entryApiCall.registration(registerRequest)
+            if(response.isSuccessful){
+                taskyCallStatus.onResponse(response.code(), response.message())
+                registrationLiveData.postValue(true)
+            }else{
+                val jObjError = response.errorBody()?.string()?.let { JSONObject(it) }
+                taskyCallStatus.onFailure(response.code(), jObjError?.getString("message") as String)
+                registrationLiveData.postValue(false)
+            }
+        }
+        return registrationLiveData
+    }
+
+    override fun doLogin(authenticationRequest: AuthenticationRequest) : LiveData<AuthenticationResponse?>{
```

Comment:
@Same comment as above regarding live data


I'd recommend just returning something like a Kotlin Result that wraps the Authentication information


Also, you are returning a Data object back to the presentation
You shouldn't have that happening

You should have:

Data model, maps to a Domain model, and use that domain model in the presentation
Iif not, the presentation is knowing stuff from the Data layer. Which shouldn't be the case

Reply:
I need a little more understanding on how this will be implemented.

Reply:
Also since what I have implemented is a skeleton code to achieve the objective of successful login and retrieving access token. The idea is to not have token here, this is just a sample code which will change in future PR.

Reply:
Also it will be great if you could direct me to some video achieving what you have explained here!!

Reply:
I don't have a video, but I made an interview couple months ago where I applied something like it
https://github.com/MKiperszmid/StoreDiscount/blob/master/app/src/main/java/com/mkiperszmid/storediscount/core/data/ProductRepositoryImpl.kt#L17

In this case, I returned a List. But you could return the object

Also, there I applied what I mentioned about mapping objects from Data to Domain
Let me know if you need more explanation :D